app/uk/gov/hmrc/economiccrimelevyreturns/controllers/StartAmendController.scala

```diff
@@ -33,76 +38,77 @@ import scala.concurrent.{ExecutionContext, Future}
 class StartAmendController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   authorise: AuthorisedAction,
-  eclAccountConnector: EclAccountConnector,
-  eclReturnsService: EclReturnsService,
+  eclAccountService: EclAccountService,
+  returnsService: ReturnsService,
   sessionService: SessionService,
-  eclReturnsConnector: EclReturnsConnector,
   noObligationForPeriodView: NoObligationForPeriodView,
   view: StartAmendView
 )(implicit ex: ExecutionContext)
     extends FrontendBaseController
-    with I18nSupport {
+    with I18nSupport
+    with BaseController
+    with ErrorHandler {
 
   def onPageLoad(periodKey: String, returnNumber: String): Action[AnyContent] = authorise.async { implicit request =>
-    for {
-      obligationData          <- eclAccountConnector.getObligations()
-      eitherObligationDetails <- validatePeriodKey(obligationData, periodKey)
-      _                        = eclReturnsService.upsertEclReturnType(request.internalId, AmendReturn)
-    } yield eitherObligationDetails match {
-      case Right(value) =>
-        val startAmendUrl = routes.StartAmendController
-          .onPageLoad(
-            periodKey = periodKey,
-            returnNumber = returnNumber
+    implicit val hc: HeaderCarrier = CorrelationIdHelper.getOrCreateCorrelationId(request)
+    (for {
+      obligationData    <- eclAccountService.retrieveObligationData.asResponseError
+      obligationDetails <- processObligationDetails(obligationData, periodKey).asResponseError
+    } yield obligationDetails).fold(
+      err => Status(err.code.statusCode)(Json.toJson(err)),
```

Comment:
How does this error message manifest? We should be showing the default error page, but this looks like it'll show JSON like what's happening in Account server.